regulation-worker/internal/service/looper.go

```diff
@@ -17,9 +17,21 @@ func (l *Looper) Loop(ctx context.Context) error {
 	for {
 		err := l.Svc.JobSvc(ctx)
 		if err == model.ErrNoRunnableJob {
-			time.Sleep(10 * time.Minute)
+			isContextCanceled := sleepContext(ctx, 10*time.Minute)
+			if isContextCanceled {
+				return nil
+			}
 		} else if err != nil {
 			return err
 		}
 	}
 }
+
+func sleepContext(ctx context.Context, delay time.Duration) bool {
```

Comment:
cool idea 🚀  Maybe we can added in a common library.

Reply:
sure

Reply:
added to utils/misc/misc.go